cmd/cstor-pool-mgmt/controller/pool-controller/after_queue_handler.go

```diff
@@ -0,0 +1,192 @@
+		if IsInitStatus(cStorPoolGot) {
+			// LabelClear is to clear pool label
+			err = pool.LabelClear(cStorPoolGot.Spec.Disks.DiskList)
+			if err != nil {
+				glog.Infof("Unable to clear pool labels : %v", err.Error())
+			}
+
+			// CreatePool is to create cstor pool.
+			err = pool.CreatePool(cStorPoolGot)
+			if err != nil {
+				return common.StatusOffline, err
+			}
+			glog.Infof("Pool creation successful")
+			return common.StatusOnline, nil
+		}
+		break
+
+	case "modify":
+		glog.Info("modify event") //ignoring as of now
+		break
+
+	case "destroy":
+		glog.Info("destroy event")
+		// DeletePool is to delete cstor pool.
+		err := pool.DeletePool("cstor-" + string(cStorPoolGot.ObjectMeta.UID))
+		if err != nil {
+			return common.StatusDeletionFailed, err
+		}
+		// removeFinalizer is to remove finalizer of cStorPool resource.
+		err = c.removeFinalizer(cStorPoolGot)
+		if err != nil {
+			return common.StatusOffline, err
+		}
+		return common.StatusIgnore, nil
+	}
+
+	return common.StatusIgnore, nil
+}
+
+// enqueueCstorPool takes a CStorPool resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorPools.
+func (c *CStorPoolController) enqueueCStorPool(obj interface{}, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.Key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// getPoolResource returns object corresponding to the resource key
+func (c *CStorPoolController) getPoolResource(key string) (*apis.CStorPool, error) {
+	// Convert the key(namespace/name) string into a distinct name
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil, nil
+	}
+
+	cStorPoolGot, err := c.clientset.OpenebsV1alpha1().CStorPools().Get(name, metav1.GetOptions{})
+	if err != nil {
+		// The cStorPool resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("cStorPoolGot '%s' in work queue no longer exists", key))
+			return nil, nil
+		}
+
+		return nil, err
+	}
+	return cStorPoolGot, nil
+}
+
+// removeFinalizer is to remove finalizer of cstorpool resource.
+func (c *CStorPoolController) removeFinalizer(cStorPoolGot *apis.CStorPool) error {
+	if len(cStorPoolGot.Finalizers) > 0 {
+		cStorPoolGot.Finalizers = []string{}
+	}
+	_, err := c.clientset.OpenebsV1alpha1().CStorPools().Update(cStorPoolGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// IsRightCStorPoolMgmt is to check if the pool request is for particular pod/application.
+func IsRightCStorPoolMgmt(cStorPool *apis.CStorPool) bool {
+	if os.Getenv("cstorid") == string(cStorPool.ObjectMeta.UID) {
+		glog.Infof("right sidecar")
+		return true
+	}
+	glog.Infof("wrong sidecar")
```

Comment:
Log message needs to be re-phrased. Also, better to make this debug log as number of occurrences will be high.

Reply:
will look into